### PR TITLE
BUG: Timestamp+int should raise NullFrequencyError, not ValueError

### LINKED ISSUE #28080: BUG: Series[int] + datetime64 should raise
identified [here](https://github.com/pandas-dev/pandas/commit/a6c0d02c99cfaf90119d1316dd897019326472c1#r34783276).  We handle this correctly for timedelta64, but apparently not datetime64

```
iser = pd.Series(range(3))
dt = np.datetime64(3, "D")

>>> iser + dt
0   1970-01-04
1   1970-01-05
2   1970-01-06
```

Should behave like `iser + pd.Timestamp(dt)`:

```
>>> iser + pd.Timestamp(dt)
[...]
ValueError: Cannot add integral value to Timestamp without freq.
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
Last prerequisite before we can fix #28080.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
What would a release note for this look like? IIUC, this is backwards compatible since catching a ValueError will still catch this NullFrequencyError? But we should still have a note describing the change so that users could tighten up the caught exception if they desire.

**Comment by jbrockmendel** [comment]
> What would a release note for this look like?

This is a good question.  I'm hesitant to advertise the change because before long we'll be enforcing the deprecation, at which point this should raise TypeError.

**Comment by WillAyd** [comment]
I think OK to just say something like `... now raises a NullFrequencyError` without worrying too much about the details of a future deprecation

File: doc/source/whatsnew/v0.25.2.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ Categorical
 
 Datetimelike
 ^^^^^^^^^^^^
-
+- Addition and subtraction of integer or integer-dtype arrays with :class:`Timestamp` will now raise ``NullFrequencyError`` instead of ``ValueError`` (:issue:`28268`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.2.rst]: move to 1.0

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.25.2.rst (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v0.25.2.rst]: changed+green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. maybe rebase and push again (as the CI is hanging)

**Comment by jbrockmendel** [comment]
rebased+green

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Looks like some merge conflicts - can you fix up and repush?

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**